testsuite/features/step_definitions/command_steps.rb

```diff
@@ -1553,6 +1546,29 @@
   raise 'Error in the output logs - see logs above' if out_spacewalk.include? 'No such file or directory'
 end
 
+When(/^I check all certificates after renaming the server hostname$/) do
+  # get server certificate serial to compare it with the other minions
+  command_server = "openssl x509 --noout --text -in /etc/pki/trust/anchors/LOCAL-RHN-ORG-TRUSTED-SSL-CERT | grep -A1 'Serial' | grep -v 'Serial'"
+  server_cert_serial, result_code = get_target('server').run(command_server)
+  server_cert_serial.strip!
+  log "Server certificate serial: #{server_cert_serial}"
+
+  raise 'Error getting server certificate serial!' unless result_code.zero?
+
+  command_minion = "openssl x509 --noout --text -in /etc/pki/trust/anchors/RHN-ORG-TRUSTED-SSL-CERT | grep -A1 'Serial' | grep -v 'Serial'"
+  targets = %w[proxy sle_minion ssh_minion rhlike_minion deblike_minion build_host kvm_server]
+  targets.each do |target|
+    # get all defined minions from the environment variables and check their certificate serial
+    next unless ENV.key? ENV_VAR_BY_HOST[target]
```

Comment:
You can either do this or:
`next if $node_by_host[target].nil?`

Reply:
Thanks for the suggestion.

Reply:
This broke the code somehow.
```bash
  Scenario: Check all new server certificates on the minions         # features/secondary/srv_rename_hostname.feature:134
      This scenario ran at: 2023-10-20 16:53:27 +0200
    When I check all certificates after renaming the server hostname # features/step_definitions/command_steps.rb:1550
      Server certificate serial: 35:a1:54:c6:5e:c3:f5:f5:c6:d8:52:e6:f9:73:20:a2:d1:c5:75:83
      FAIL: openssl x509 --noout --text -in /etc/pki/trust/anchors/RHN-ORG-TRUSTED-SSL-CERT | grep -A1 'Serial' | grep -v 'Serial' returned status code = 1.
      Output:
      Can't open /etc/pki/trust/anchors/RHN-ORG-TRUSTED-SSL-CERT for reading, No such file or directory
      139797779679040:error:02001002:system library:fopen:No such file or directory:crypto/bio/bss_file.c:69:fopen('/etc/pki/trust/anchors/RHN-ORG-TRUSTED-SSL-CERT','r')
      139797779679040:error:2006D080:BIO routines:BIO_new_file:no such file:crypto/bio/bss_file.c:76:
      unable to load certificate
       (RuntimeError)
      ./features/support/lavanda.rb:177:in `run_local'
      ./features/support/lavanda.rb:155:in `run'
      ./features/step_definitions/command_steps.rb:1561:in `block (2 levels) in <top (required)>'
      ./features/step_definitions/command_steps.rb:1560:in `each'
      ./features/step_definitions/command_steps.rb:1560:in `/^I check all certificates after renaming the server hostname$/'
      features/secondary/srv_rename_hostname.feature:135:in `I check all certificates after renaming the server hostname'
=> /var/log/rhn/rhn_web_ui.log
```